spharpy/spherical.py

```diff
@@ -1005,3 +1007,413 @@ def sph_identity_matrix(n_max, type='n-nm'):
         identity_matrix[n, linear_nm] = 1
 
     return identity_matrix
+
+
+
+class SphericalHarmonics:
+    r"""
+    Compute spherical harmonic basis matrices, their inverses, and gradients.
+    The spherical harmonic Ynm is given by:
+    .. math::
+        Y_{nm} = N_{nm} P_{nm}(cos(\theta)) T_{nm}(\phi)
+
+    where:
+    - :math:`n` is the degree
+    - :math:`m` is the order
+    - :math:`P_{nm}` is the associated Legendre function
+    - :math:`N_{nm}` is the normalization term
+    - :math:`T_{nm}` is a term that depends on whether the
+     harmonics are real or complex
+    - :math:`\theta` is the colatitude (angle from the positive z-axis)
+    - :math:`\phi` is the azimuth (angle in the x-y plane from the x-axis)
+
+    The normalization term :math:`N_{nm}` is given by:
+
+    .. math::
+        N_{nm}^{\text{SN3D}} =
+        \sqrt{\frac{2n+1}{4\pi} \frac{(n-|m|)!}{(n+|m|)!}}
+
+        N_{nm}^{\text{N3D}} = N_{nm}^{\text{SN3D}} \sqrt{\frac{2n+1}{2}}
+
+        N_{nm}^{\text{MaxN}} = ... (max of N3D)
+
+    The associated Legendre function :math:`P_{nm}` is defined as:
+
+    .. math::
+        P_{nm}(x) = (1-x^2)^{|m|/2} (d/dx)^n (x^2-1)^n
+
+    The term :math:`T_{nm}` is defined as:
+        - For complex-valued harmonics:
+            .. math::
+                T_{nm} = e^{im\phi}
+        - For real-valued harmonics:
+            .. math::
+                T_{nm} = \begin{cases}
+                            \cos(m\phi) & \text{if } m \ge 0 \\
+                            \sin(m\phi) & \text{if } m < 0
+                        \end{cases}
```

Comment:
```suggestion

    - For complex-valued harmonics:
        .. math::
            T_{nm} = e^{im\phi}
    - For real-valued harmonics:
        .. math::
            T_{nm} = \begin{cases}
                        \cos(m\phi) & \text{if } m \ge 0 \\
                        \sin(m\phi) & \text{if } m < 0
                    \end{cases}
```
i suggest to remove the intend, then it looks more pretty in the doc and in the docstring.